### ISSUE TITLE
ExplicitlyDefineReturn

### ISSUE THREAD

**Issue by mustafaozhan** [detekt/detekt#5692]
## Expected Behavior of the rule
When a method returns via `=`, and the result is not `Unit` warn that return type should be explicitly defined.

## Context
One of the cool thing with Kotlin having methods without bracket and return the result with `=` but it causes readability issues especially if the method is complex.

This rule should simply warn developers to use

```kotlin
fun getCalculationResult(): Double = Calculator.calculate()
```
instead of
```kotlin
fun getCalculationResult() = Calculator.calculate()
```

where it doesn't warn
```kotlin
fun justPrintHello() = println("hello")
```

**Comment by BraisGabin** [comment]
I think that your rule request have a lot of sense. We already have a rule for the case that you say that shouldn't warn: `ImplicitUnitReturnType`.

I'm not sure if it have sense to merge this idea with that rule and create a new one (deprecation the old one) or keep them as two different rules.

As a side note: I usually use the [explicit api mode](https://kotlinlang.org/docs/whatsnew14.html#explicit-api-mode-for-library-authors) that force me to define it for all my `public` classes.

**Comment by mustafaozhan** [comment]
@BraisGabin glad that you like it 🙂

Yes, I tried both `explicit api mode` and `ImplicitUnitReturnType` and couldn't get what I describe in the issue.

`explicit api mode` annoys with forcing you to use `Unit` as well, and I couldn't configure it. Also, when I try last time, there was an inconsistent behaviour for the method which has both visibility and return type issue, I can not recall exactly now but when you fix one of them the other one was disappearing.

Unfortunately, I do not have much free time for a couple of months and I think someone already familiar to the project will get the job done much faster than me.

**Comment by atulgpt** [comment]
Hi @BraisGabin I can take this up if still available

**Comment by BraisGabin** [comment]
Sorry for the late answer. Sure, this is available. I think that we should merge this PR with `ImplicitUnitReturnType`. Probably the rule can be called `ExplicitlyDefineReturn` and a flag `allowOmitUnit`.

**Comment by atulgpt** [comment]
Hi, @BraisGabin I have added the rule `ExplicitlyDefineReturn`  but have not merged it with `ImplicitUnitReturnType`. To me, it seemed that `ImplicitUnitReturnType` more focuses on avoiding expression statements for only `Unit` returning statement(and hence very less scope) while `ExplicitlyDefineReturn` tries to mimic `explicitApiMode = "strict"` with some relaxation.

Using `ImplicitUnitReturnType` I can force `fun log(msg) = println(msg)` to be written as `fun log(msg) { println(msg) }` but not with `ExplicitlyDefineReturn`
